stages/org.osbuild.zipl-secex.inst

```diff
@@ -0,0 +1,234 @@
+within the device that is specified via `disk`.
+- `ignition_gpg_pubkey`, which specifies filename for public GPG key
+- `hostkey`, which specifies Hostkey for given IBM Z machine
+- `build_id`, which specifies CoreOS build
+"""
+
+import libarchive
+import os
+import subprocess
+import sys
+import tempfile
+
+import osbuild.api
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "required": ["kernel", "location", "ignition_gpg_pubkey", "hostkey", "build_id"],
+  "properties": {
+    "kernel": {
+      "type": "string"
+    },
+    "location": {
+      "type": "number"
+    },
+    "hostkey": {
+      "type": "string"
+    },
+    "ignition_gpg_pubkey": {
+       "type": "string"
+    },
+    "build_id": {
+       "type": "string"
+    },
+    "kernel_opts_append": {
+      "description": "Additional kernel options to add to the discovered options",
+      "type": "array",
+      "items": {
+        "description": "A single kernel command line option",
+        "type": "string"
+      }
+    },
+    "blocksize": {
+      "type": "number",
+      "default": 512
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true,
+  "required": ["disk"],
+  "properties": {
+    "disk": {
+      "type": "object",
+      "additionalProperties": true
+    }
+  }
+},
+"mounts": {
+  "type": "array",
+  "minItems": 1
+}
+"""
+
+def parse_blsfile(blsfile):
+    params = {}
+    with open(blsfile, "r", encoding="utf8") as bls:
+        for line in bls:
+            key, value = line.split(' ', 1)
+            params[key] = value.strip()
+    return params
+
+
+def find_kernel(root, kernel: str):
+    base = f"{root}/boot/loader/entries"
+    for dirent in os.scandir(base):
+        fn, ext = os.path.splitext(dirent.name)
+        if ext != ".conf" or fn.endswith("rescue"):
+            continue
+        blsfile = f"{base}/{dirent.name}"
+        params = parse_blsfile(blsfile)
+
+        version = params["version"]
+        if version != kernel:
+            continue
+
+        linux = root + params["linux"]
+        initrd = root + params["initrd"]
+        options = params.get("options", "")
+        return linux, initrd, options
+
+    raise ValueError(f"Unknown kernel: {kernel}")
+
+
+def generate_gpg_keys(ignition_pubkey, build_id):
+    with tempfile.TemporaryDirectory() as d:
+      subprocess.run(["gpg",
+                      "--homedir", d,
+                      "--batch",
+                      "--passphrase", '',
+                      "--yes",
+                      "--quick-gen-key", f"Secure Execution (secex) {build_id}",
+                      "rsa4096",
+                      "encr",
+                      "none"],
+                     check=True)
+      subprocess.run(["gpg",
+                      "--homedir", d,
+                      "--output", ignition_pubkey,
+                      "--armor",
+                      "--export", "secex"],
+                     check=True)
```

Comment:
is this exporting the pubkey of the key that was just generated in the previous step? I'm trying to understand how just specifying `secex` here does that where above we specify a fingerprint of `"Secure Execution (secex) {build_id}"`.

Reply:
Yes, `secex` is enough to find the key (we are in temp folder, so there is just 1 GPG key).